File: server/models/VisitorLog.js
const mongoose = require('mongoose');

const VisitorLogSchema = new mongoose.Schema({
  ip: String,
  page: String,
  action: String, // 'Visit', 'Click', etc.
  userAgent: String,
  timestamp: { type: Date, default: Date.now }
});

module.exports = mongoose.model('VisitorLog', VisitorLogSchema);

File: backend/src/main/java/com/portfolio/entity/VisitorLog.java
package com.portfolio.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Getter @Setter
@Table(name = "visitor_logs")
public class VisitorLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String ip;
    private String page;
    private String action; // 'Visit', 'Click', etc.
    private String userAgent;

    @CreationTimestamp
    private LocalDateTime timestamp;
}
